Doc/whatsnew/3.14.rst

```diff
@@ -1396,6 +1396,7 @@ Deprecated
   * :c:func:`!_PyDict_GetItemStringWithError`: use :c:func:`PyDict_GetItemStringRef`.
   * :c:func:`!_PyDict_Pop()`: use :c:func:`PyDict_Pop`.
   * :c:func:`!_PyLong_Sign()`: use :c:func:`PyLong_GetSign`.
+  * :c:func:`!_PyLong_New`: use :c:func:`PyLongWriter_Create`.
```

Comment:
Oh, you should also add the function to Doc/deprecations/c-api-pending-removal-in-3.18.rst.